### PR TITLE
Upgrade to target-based CMake. Added option for shared libraries.

### PR DESCRIPTION
**Pull request by psalvaggio**

### CODE REVIEW

**Comment by psalvaggio** [comment]
The Travis failures seem to have to deal with a difference in Boost versions. It looks like Boost 1.54.0 doesn't have the CMake targets defined. I can switch that easily enough. The AppVeyor tests, on the other hand, I'm not so sure what the issue actually is.

**Comment by nekipelov** [comment]
Looks good. Can you fix Windows build?

**Comment by psalvaggio** [comment]
I'll give it a shot. I have basically no Visual Studio experience (or Visual Studio for that matter). I'll see if I can reproduce the error in the free version.

**Comment by psalvaggio** [comment]
OK, so it looks like it was expecting static libraries instead of DLLs, I added the static library option in the appveyor.yml file and it seems to work.

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -1,49 +1,21 @@
...
-)
-
-SET(EXAMPLES
+set(EXAMPLES
     async_pubsub2.cpp
     async_pubsub.cpp
     async_set_get2.cpp
     async_set_get.cpp
     sync_set_get.cpp
-    sync_pipeline.cpp
```
Comment (nekipelov) [inline examples/CMakeLists.txt]: Why did you delete this line?

**Comment by psalvaggio** [comment]
Ahh, I forgot about that one, I was getting a couple of compile errors about redisclient::Pipeline being an incomplete type. I'll go back and fix that.

-Phil

**Comment by nekipelov** [comment]
Thank you!
